VKappProject/VKappProject/Controllers/AvailableGroupsTableViewController.swift

```diff
@@ -70,27 +57,21 @@ final class AvailableGroupsTableViewController: UITableViewController {
                     .availableGroupsCellIdentifier
             ) as? AvailableGroupTableViewCell
         else { return UITableViewCell() }
-        let group = isFiltering ? searchResult[indexPath.row] : availableGroups[indexPath.row]
+        let group = searchResult[indexPath.row]
         cell.configureCell(group)
         return cell
     }
-
-    override func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
-        chooseGroup(indexPath)
-    }
 }
 
 // MARK: - UISearchBarDelegate
 
 extension AvailableGroupsTableViewController: UISearchBarDelegate {
     func searchBar(_ searchBar: UISearchBar, textDidChange searchText: String) {
-        searchResult = availableGroups.filter { $0.groupName.lowercased().contains(searchText.lowercased()) }
-        isSearching = true
+        fetchSearchedGroupsRequest(searchedText: searchText)
         tableView.reloadData()
     }
 
     func searchBarCancelButtonClicked(_ searchBar: UISearchBar) {
-        isSearching = false
         searchBar.text = ""
```

Comment:
```suggestion
        searchBar.text = Constants.emptyString
```
Меня заставляли выносить в константы